src/app/page/component/page/cyoa-card-collection-page/cyoa-card-collection-page.component.ts

```diff
@@ -0,0 +1,179 @@
+  CyoaCardCollectionPage
+} from 'src/app/services/xml-parser-service/xmp-parser.service';
+import { PageService } from '../../../service/page-service.service';
+
+@Component({
+  selector: 'app-cyoa-card-collection-page',
+  templateUrl: './cyoa-card-collection-page.component.html',
+  styleUrls: ['../default-page.css'],
+  encapsulation: ViewEncapsulation.None
+})
+export class CYOACardCollectionComponent implements OnChanges, OnDestroy {
+  @Input() page: CyoaCardCollectionPage;
+  @Input() order: number;
+  @Input() totalPages: number;
+
+  readonly _unsubscribeAll: Subject<void>;
+  private _page: CyoaCardCollectionPage;
+  cards: CYOAPageCard[];
+  ready: boolean;
+  dir$: Observable<string>;
+  formAction$: Observable<string>;
+  isForm$: Observable<boolean>;
+  isModal$: Observable<boolean>;
+  currentYear = new Date().getFullYear();
+  isLastCard: boolean;
+  isFirstCard: boolean;
+  totalCards: number;
+  showBackButton: boolean;
+
+  constructor(
+    readonly pageService: PageService,
+    private router: Router,
+    private route: ActivatedRoute
+  ) {
+    this._unsubscribeAll = new Subject<void>();
+    this.dir$ = this.pageService.pageDir$;
+    this.isForm$ = this.pageService.isForm$;
+    this.isModal$ = this.pageService.isModal$;
+    this.formAction$ = this.pageService.formAction$;
+  }
+
+  ngOnDestroy() {
+    this._unsubscribeAll.next();
+    this._unsubscribeAll.complete();
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    for (const propName in changes) {
+      if (changes.hasOwnProperty(propName)) {
+        switch (propName) {
+          case 'page':
+            if (
+              !changes['page'].previousValue ||
+              changes['page'].currentValue !== changes['page'].previousValue
+            ) {
+              this.ready = false;
+              this._page = this.page;
+              this.cards = [];
+              this.init();
+            }
+            break;
+          default:
+            break;
+        }
+      }
+    }
+  }
+  currentCardIndex = 0;
+
+  get currentCard(): CYOAPageCard | undefined {
+    return this.cards?.[this.currentCardIndex];
+  }
+
+  goToCard(index: number): void {
+    if (!this.cards || this.cards.length === 0) return;
+
+    const safeIndex = Math.max(0, Math.min(index, this.cards.length - 1));
+
+    if (safeIndex !== this.currentCardIndex) {
+      this.currentCardIndex = safeIndex;
+      this.updateUrl();
+      this.updateCardState();
+    }
+  }
+
+  showPreviousCard(): void {
+    this.goToCard(this.currentCardIndex - 1);
+  }
+
+  showNextCard(): void {
+    this.goToCard(this.currentCardIndex + 1);
+  }
+
+  private updateUrl(): void {
+    this.router.navigate(['../', this.currentCardIndex], {
+      relativeTo: this.route
+    });
+  }
+
+  private updateCardState(): void {
+    this.isFirstCard = this.currentCardIndex === 0;
+    this.isLastCard = this.currentCardIndex === this.totalCards - 1;
+  }
+
+  private subscribeToCardPosition(): void {
```

Comment:
We normally name this with `on` at the front of the function name.
For example: `onCardPositionChange`